Lib/inspect.py

```diff
@@ -1790,9 +1790,6 @@ def _signature_get_partial(wrapped_sig, partial, extra_args=()):
             if param.kind is _POSITIONAL_OR_KEYWORD:
                 new_param = new_params[param_name].replace(kind=_KEYWORD_ONLY)
                 new_params[param_name] = new_param
-                new_params.move_to_end(param_name)
```

Comment:
Let's keep the "move_to_end" behavior.  The same effect is easy to achieve with regular dicts.


```
   -            new_params[param_name] = new_param
   -            new_params.move_to_end(param_name)
   +            new_params.pop(param_name, None)
   +            new_params[param_name] = new_param
```

Reply:
Hi @rhettinger, do you want to keep the behavior to show it explicitly or because it's necessary? I'm not sure it's needed (https://github.com/python/cpython/pull/12412#discussion_r266526157) but I'm having a hard time wrapping my head around this.

Reply:
Because we want to avoid introducing behavior changes.  Someone wrote this code on purpose.  The goal here is to just substitute a smaller, faster dict, so let's focus on just that.

Reply:
Thanks I restablished the previous behavior.